scimono-client/src/main/java/com/sap/scimono/client/SCIMClientService.java

```diff
@@ -96,8 +98,15 @@ public Builder addProperty(String name, Object value) {
       properties.put(name, value);
       return this;
     }
+    
+    public Builder setUserNameOptional(boolean isOptional) {
+      isUserNameOptional = isOptional;
+      return this;
+    }
```

Comment:
As we talked internally with Vladi. I think that here is not the place where such a configuration must be done. 
It will be more appropriate if such can be placed where the actual `buildUserRequest(...)` method is used. 
However, if this cannot be achieved easily (lead to a lot refactoring activities), it will be better if we add more generic structure for such configurations (e.g  `addUserPropertiesConfiguration(....).setUserNameOptional()`